shared/src/shared.rs

```diff
@@ -81,13 +82,14 @@ impl<CI: ChainIndex> Shared<CI> {
                 proposal_ids,
                 TxPool::new(tx_pool_config),
                 LruCache::new(txs_verify_cache_size),
+                Arc::clone(&consensus),
```

Comment:
I think that we can refactor ChainState construct function now:
```rust
ChainState::new(&store, tx_pool_config, Arc::clone(&consensus))
```

and move all init code to construct function.